async-nats/src/lib.rs

```diff
@@ -338,6 +337,12 @@ impl ConnectionHandler {
         mut receiver: mpsc::Receiver<Command>,
     ) -> Result<(), io::Error> {
         loop {
+            let flush_sleep = if self.flush_wanted {
+                tokio::time::sleep(self.flush_period).boxed()
```

Comment:
This will create a lot of allocator noise. What if we made the `Interval` very high (a few days long) when we don't want any flushes, and lowered it back to the configured value when we want it back?

Reply:
Or we could make our own `Interval` wrapper around [`Sleep`](https://docs.rs/tokio/latest/tokio/time/struct.Sleep.html#method.reset)

> This function can be called both before and after the future has completed.

Reply:
> This will create a lot of allocator noise.
Aye, but we can take it out again with a little bit of work.

> Or we could make our own Interval wrapper around [Sleep](https://docs.rs/tokio/latest/tokio/time/struct.Sleep.html#method.reset)

Yeah short rundown is that we'd want to return pending when flush not wanted, store the waker, wake up when flush wanted is.